sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangePartitionKeyImpl.java

```diff
@@ -30,73 +37,125 @@ public PartitionKeyInternal getPartitionKeyInternal() {
+
+                Range<String> range = Range.getPointRange(effectivePartitionKey);
+                return Mono.just(range);
+            });
     }
 
     @Override
-    public Mono<UnmodifiableList<String>> getPartitionKeyRanges(
+    public Mono<List<String>> getPartitionKeyRanges(
         IRoutingMapProvider routingMapProvider,
-        String containerRid,
-        PartitionKeyDefinition partitionKeyDefinition) {
-
-        String effectivePartitionKey = this.partitionKey.getEffectivePartitionKeyString(
-            this.partitionKey,
-            partitionKeyDefinition);
-        return routingMapProvider
-            .tryGetOverlappingRangesAsync(
-                null,
-                containerRid,
-                Range.getPointRange(effectivePartitionKey),
-                false,
-                null)
-            .flatMap(pkRangeHolder -> {
-                ArrayList<String> rangeList = new ArrayList<>();
-
-                if (pkRangeHolder != null) {
-                    String rangeId = pkRangeHolder.v.get(0).getId();
-                    rangeList.add(rangeId);
+        RxDocumentServiceRequest request,
+        Mono<Utils.ValueHolder<DocumentCollection>> collectionResolutionMono) {
+
+        checkNotNull(
+            routingMapProvider,
+            "Argument 'routingMapProvider' must not be null");
+        checkNotNull(
+            request,
+            "Argument 'request' must not be null");
+        checkNotNull(
+            collectionResolutionMono,
+            "Argument 'collectionResolutionMono' must not be null");
+
+        MetadataDiagnosticsContext metadataDiagnosticsCtx =
+            BridgeInternal.getMetaDataDiagnosticContext(request.requestContext.cosmosDiagnostics);
+
+        return collectionResolutionMono
+            .flatMap(documentCollectionResourceResponse -> {
+
+                final DocumentCollection collection = documentCollectionResourceResponse.v;
+                if (collection == null) {
+                    throw new IllegalStateException("Collection cannot be null");
                 }
 
-                return Mono.just((UnmodifiableList<String>)UnmodifiableList.unmodifiableList(rangeList));
+                final String containerRid = collection.getResourceId();
+                final String effectivePartitionKey =
+                    this.partitionKey.getEffectivePartitionKeyString(
+                    this.partitionKey,
+                    collection.getPartitionKey());
+
+                return routingMapProvider
+                    .tryGetOverlappingRangesAsync(
+                        metadataDiagnosticsCtx,
+                        containerRid,
+                        Range.getPointRange(effectivePartitionKey),
+                        false,
+                        null)
+                    .flatMap(pkRangeHolder -> {
+                        ArrayList<String> rangeList = new ArrayList<>();
+
+                        if (pkRangeHolder != null) {
```

Comment:
as per reactive-stream contract, null value is not a permissible value hence this will be always true,
perhaps you should check
`pkRangeHolder.v != null` instead

Reply:
Fixed in next iteration